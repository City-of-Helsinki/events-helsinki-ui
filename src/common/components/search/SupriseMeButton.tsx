import React, { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";

import { ReactComponent as SupriseMeIcon } from "../../../assets/icons/svg/suprise-me.svg";
import styles from "./supriseMeButton.module.scss";

interface Props {
  onClick: () => void;
}

const SupriseMeButton: FunctionComponent<Props> = ({ onClick }) => {
  const { t } = useTranslation();

  return (
    <button className={styles.supriseMeButton} onClick={onClick} type="button">
      <div className={styles.text}>{t("commons.supriseMe")}</div>
      <SupriseMeIcon />
    </button>
  );
};

export default SupriseMeButton;
