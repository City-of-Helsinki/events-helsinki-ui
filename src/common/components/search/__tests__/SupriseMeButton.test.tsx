import { render } from '@testing-library/react';
import * as React from 'react';

import SupriseMeButton from '../SupriseMeButton';

test('SupriseMeButton matches snapshot', () => {
  const { container } = render(<SupriseMeButton onClick={jest.fn()} />);

  expect(container.firstChild).toMatchSnapshot();
});
