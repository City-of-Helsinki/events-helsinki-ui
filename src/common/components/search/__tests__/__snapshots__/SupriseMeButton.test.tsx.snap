// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupriseMeButton matches snapshot 1`] = `
<button
  className="supriseMeButton"
  onClick={[Function]}
  type="button"
>
  <div
    className="text"
  >
    Koeta onneasi
  </div>
  <svg>
    suprise-me.svg
  </svg>
</button>
`;
